homeassistant/components/squeezebox/__init__.py

```diff
@@ -117,15 +128,54 @@ async def async_setup_entry(hass: HomeAssistant, entry: SqueezeboxConfigEntry) -
     )
     _LOGGER.debug("LMS Device %s", device)
 
-    coordinator = LMSStatusDataUpdateCoordinator(hass, lms)
+    server_coordinator = LMSStatusDataUpdateCoordinator(hass, lms)
 
     entry.runtime_data = SqueezeboxData(
-        coordinator=coordinator,
+        coordinator=server_coordinator,
         server=lms,
     )
 
-    await coordinator.async_config_entry_first_refresh()
+    # set up player discovery
+    known_servers = hass.data.setdefault(DOMAIN, {}).setdefault(KNOWN_SERVERS, {})
+    known_players = known_servers.setdefault(lms.uuid, {}).setdefault(KNOWN_PLAYERS, [])
+
+    async def _player_discovery(now: datetime | None = None) -> None:
+        """Discover squeezebox players by polling server."""
+
+        async def _discovered_player(player: Player) -> None:
+            """Handle a (re)discovered player."""
+            if player.player_id in known_players:
+                async_dispatcher_send(
+                    hass, SIGNAL_PLAYER_REDISCOVERED, player.player_id, player.connected
+                )
+            else:
+                _LOGGER.debug("Adding new entity: %s", player)
+                player_coordinator = SqueezeBoxPlayerUpdateCoordinator(
+                    hass, player, lms.uuid
+                )
+                known_players.append(player.player_id)
+                async_dispatcher_send(
+                    hass, SIGNAL_PLAYER_DISCOVERED, player_coordinator
+                )
+
+        if players := await lms.async_get_players():
+            for player in players:
+                hass.async_create_task(_discovered_player(player))
+
+        entry.async_on_unload(
+            async_call_later(hass, DISCOVERY_INTERVAL, _player_discovery)
```

Comment:
Just to verify, this is meant to start discovery after it has started right?

Reply:
We need to periodically poll the server for players. What this does is add another poll at DISCOVERY_INTERVAL into the future after the current poll completes.

Reply:
I don't think it does that. Async call later is "call function X after Y seconds" without any repeats

Reply:
That's right. But this is the end of the player discovery function. Its last act is to queue itself up again after DISCOVERY_INTERVAL seconds.